oci/layout/oci_delete_test.go

```diff
@@ -0,0 +1,254 @@
+
+	sys := &types.SystemContext{OCISharedBlobDirPath: filepath.Join(tmpDir, "shared_blobs")}
+	err = ref.DeleteImage(context.Background(), sys)
+	require.NoError(t, err)
+
+	// Check that the only blob in the local directory was deleted
+	blobsDir := filepath.Join(tmpDir, "blobs")
+	files, err := os.ReadDir(filepath.Join(blobsDir, "sha256"))
+	require.NoError(t, err)
+	require.Empty(t, files)
+
+	// Check that the blobs in the shared blob directory are still present
+	sharedBlobsDir := filepath.Join(tmpDir, "shared_blobs")
+	files, err = os.ReadDir(filepath.Join(sharedBlobsDir, "sha256"))
+	require.NoError(t, err)
+	require.Equal(t, 2, len(files))
+
+	// Check that the index is empty as there is only one image in the fixture
+	ociRef, ok := ref.(ociReference)
+	require.True(t, ok)
+	index, err := ociRef.getIndex()
+	require.NoError(t, err)
+	require.Equal(t, 0, len(index.Manifests))
+}
+
+func TestReferenceDeleteImage_multipleImages(t *testing.T) {
+	tmpDir := loadFixture(t, "delete_image_multiple_images")
+
+	ref, err := NewReference(tmpDir, "3.17.5")
+	require.NoError(t, err)
+
+	err = ref.DeleteImage(context.Background(), nil)
+	require.NoError(t, err)
+
+	// Check that the relevant blobs were deleted/preservend
+	blobsDir := filepath.Join(tmpDir, "blobs")
+	files, err := os.ReadDir(filepath.Join(blobsDir, "sha256"))
+	require.NoError(t, err)
+	require.Equal(t, 16, len(files))
+	blobDoesNotExist(t, blobsDir, "sha256:5b2aba4d3c27bc6493633d0ec446b25c8d0a5c9cfe99894bcdff0aee80813805")
+	blobDoesNotExist(t, blobsDir, "sha256:df11bc189adeb50dadb3291a3a7f2c34b36e0efdba0df70f2c8a2d761b215cde")
+	blobDoesNotExist(t, blobsDir, "sha256:986315a0e599fac2b80eb31db2124dab8d3de04d7ca98b254999bd913c1f73fe")
+
+	// Check the index
+	ociRef, ok := ref.(ociReference)
+	require.True(t, ok)
+	// .. Check that the index has been reduced to the correct size
+	index, err := ociRef.getIndex()
+	require.NoError(t, err)
+	require.Equal(t, 6, len(index.Manifests))
+	// .. Check that the image is not in the index anymore
+	for _, descriptor := range index.Manifests {
+		switch descriptor.Annotations[imgspecv1.AnnotationRefName] {
+		case "3.17.5":
+			assert.Fail(t, "image still present in the index after deletion")
+		default:
+			continue
+		}
+	}
+}
+
+func TestReferenceDeleteImage_multipleImages_blobsUsedByOtherImages(t *testing.T) {
+	tmpDir := loadFixture(t, "delete_image_multiple_images")
+
+	ref, err := NewReference(tmpDir, "1.0.0")
+	require.NoError(t, err)
+
+	err = ref.DeleteImage(context.Background(), nil)
+	require.NoError(t, err)
+
+	// Check that the relevant blobs were deleted/preservend
+	blobsDir := filepath.Join(tmpDir, "blobs")
+	files, err := os.ReadDir(filepath.Join(blobsDir, "sha256"))
+	require.NoError(t, err)
+	require.Equal(t, 17, len(files))
+	blobExists(t, blobsDir, "sha256:df11bc189adeb50dadb3291a3a7f2c34b36e0efdba0df70f2c8a2d761b215cde")
+	blobDoesNotExist(t, blobsDir, "sha256:0dc27f36a618c110ae851662c13283e9fbc1b5a5de003befc4bcefa5a05d2eef")
```

Comment:
Absolutely non-blocking: Naming these helpers `assertBlob…` would be a tiny bit more readable, indicating the responsibility to assert, not just to return a boolean value as a predicate-named function typically does.